### PR TITLE
76 : PluginManager 

### PR DESCRIPTION
**Pull request by hbaptiste**
New PluginManager that will allow us to create plugins for contents.
A plugin can expose 0 to n actions that a ContentActionWidget will show above the selected content zone.
Exposing actions is not mandatory. Plugin can also modify contents in many different ways.

### CODE REVIEW

File: src/tb/apps/content/components/ContentActionWidget.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+define(['jquery', 'text!content-templates/content-action.twig', 'jsclass'], function (jQuery, template) {
+    'use strict';
+    var ContentActionWidget = new JS.Class({
+        initialize: function () {
+            this.content = null;
+            this.widget = jQuery(template).clone();
+            jQuery(this.widget).css({
```
Comment (ndufreche) [inline src/tb/apps/content/components/ContentActionWidget.js]: Plz no css in the javascript add a class.

File: src/tb/apps/content/components/ContentActionWidget.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    return function () {
+                        actionInfos.cmd.execute();
+                    };
+                }(actionInfos)));
+                btnCtn.appendChild(jQuery(button).get(0));
+            });
+            this.widget.append(btnCtn);
+        },
+        cleanActions: function () {
+            //jQuery(this.content).find(this.widget).remove();
```
Comment (ndufreche) [inline src/tb/apps/content/components/ContentActionWidget.js]: Remove commented code

File: src/tb/apps/content/components/ContentManager.js [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+                /**
+                 * Retrieve a object identifier for split uid and type
+                 */
+                retrievalObjectIdentifier: function (objectIdentifier) {
+                    var regex,
+                        res = {};
+
+                    if (objectIdentifier) {
+                        /*jslint regexp: true */
+                        regex = /(.+)\(([a-f0-9]+)\)$/;
```
Comment (ndufreche) [inline src/tb/apps/content/components/ContentManager.js]: What kind of identifier this regex have to parse ?

**Comment by mickaelandrieu** [comment]
Why you pull request fix multiples tickets ? Did you have rebased master branch ?

File: src/tb/apps/content/components/ContentManager.js [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+                 */
+                dropAsHtml: function (config, html) {
+                    if (config.parent) {
+                        config.parent.prepend(html);
+                    } else if (config.prevSibling) {
+                        config.prevSibling.after(html);
+                    }
+                },
+
+                /***** EVENTS *****/
```
Comment (ndufreche) [inline src/tb/apps/content/components/ContentManager.js]: It seems that your class do to much work.

File: src/tb/apps/content/components/ContentManager.js [hunk 5]
```diff
@@ -0,0 +1,495 @@
+/*
```
Comment (ndufreche) [inline src/tb/apps/content/components/ContentManager.js]: This file is to much specific to be a part of a component

File: src/tb/apps/content/plugins/contenttype.plugin.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
+define(['content.pluginmanager', 'jquery', 'jsclass'], function (PluginManager, jQuery) {
+    'use strict';
+    PluginManager.registerPlugin('contenttype', {
+        ACCEPT_CONTENT_TYPE: ["Bloc/contentset_bfm", 'home/home_container'],
```
Comment (ndufreche) [inline src/tb/apps/content/plugins/contenttype.plugin.js]: Please remove bfm reference

File: src/tb/core/Mediator.js [hunk 7]
```diff
@@ -51,7 +51,6 @@ define('tb.core.Mediator', ['tb.core.Api'], function (Api) {
      */
```
Comment (ndufreche) [inline src/tb/core/Mediator.js]: Why there is a commit on Mediator ?
Please check what you commit.

**Comment by ndufreche** [comment]
This PR is too large, broke the build and resolve multiple ticket.

Please resolve all this problems.

**Pull request closed without merging**